pkg/controller/controller.go

```diff
@@ -84,9 +89,18 @@ func (c *Controller) handleClusterEvent(event *Event) (bool, error) {
 			}
 			delete(c.clusters, getNamespacedName(clus))
 
+			// operator does not clean up resources on it's own, it relies
+			// on ownerReferences / k8s gc to cleanup orphaned pods
+			// for this case we want to clean up manually
+			c.logger.Info("cleaning up cluster resources")
+			err := c.cleanupClusterResources(clus)
+			if err != nil {
+				c.logger.Errorf("unable to cleanup cluster resources: %v", err)
+			}
+
```

Comment:
This is the logical change: cleans up pods, peer-service and client-service of this particular cluster.